content/javascript/concepts/promise/terms/all/all.md

```diff
@@ -23,39 +31,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
 
 ## Example
 
-Working with two Promise objects, `promiseA` and `promiseB`:
+Working with multiple Promise objects to perform parallel API requests:
 
 ```js
-const promiseA = new Promise((resolve, reject) => {
-  resolve(23);
-});
-
-const promiseB = new Promise((resolve, reject) => {
-  if (12 * 12 === 144) {
-    resolve(144);
-  } else {
-    reject({
-      errorType: 'TypeError',
-      message: `Unexpected type - expected ${typeof (12 * 12)}.`,
-    });
-  }
-});
-
-Promise.all([promiseA, promiseB])
-  .then((values) => {
-    console.log(`Results from Promise.all(): [${values}]`);
-  })
-  .catch((err) => {
-    console.log(`Promise.all Failed! \n${err.errorType}: ${err.message}`);
+const fetchUserData = () => {
+  return new Promise((resolve) => {
+    // Simulating API call
+    setTimeout(() => {
+      const userData = { id: 1, name: 'John Doe', email: 'john.doe@example.com' };
+      resolve(userData);
+    }, 2000);
+  });
+};
+
+const fetchProductData = () => {
+  return new Promise((resolve) => {
+    // Simulating API call
+    setTimeout(() => {
+      const productData = { id: 123, name: 'Example Product', price: 19.99 };
+      resolve(productData);
+    }, 1500);
+  });
+};
+
+const fetchOrderData = () => {
+  return new Promise((resolve) => {
+    // Simulating API call
+    setTimeout(() => {
+      const orderData = { id: 456, products: ['Example Product'], total: 19.99 };
+      resolve(orderData);
+    }, 1000);
+  });
+};
+
+Promise.all([fetchUserData(), fetchProductData(), fetchOrderData()])
+  .then((results) => {
+    const [userData, productData, orderData] = results;
+    console.log('User Data:', userData);
+    console.log('Product Data:', productData);
+    console.log('Order Data:', orderData);
   })
-  .finally(() => {
-    console.log('Operations for Promise.all() have finished.');
+  .catch((error) => {
+    console.log('Error occurred:', error);
   });
+
 ```
+## Explanation
+
+In this example, we have three functions (fetchUserData(), fetchProductData(), and fetchOrderData()) that simulate API calls with a delay using setTimeout(). Each function returns a Promise that resolves with the respective data.
+
+We use Promise.all() to execute these promises in parallel and wait for all of them to resolve. The .then() callback receives an array of resolved values in the same order as the input promises. We destructure the array to extract the user data, product data, and order data, and log them to the console.
```

Comment:
[Proofreading/Nipick] Refer to [my comment on line 5](https://github.com/Codecademy/docs/pull/2459/files#r1225974895). It shares the same issue as this line.  
```suggestion
In this example, we have three functions (fetchUserData(), fetchProductData(), and fetchOrderData()) that simulate API calls with a delay using setTimeout(). Each function returns a Promise that resolves with the respective data.

We use Promise.all() to execute these promises in parallel and wait for all of them to resolve. The .then() callback receives an array of resolved values in the same order as the input promises. We destructure the array to extract the user data, product data, and order data, and log them to the console.
```